src/sparseml/pytorch/sparsification/pruning/modifier_pruning_mfac.py

```diff
@@ -259,14 +272,21 @@ def initialize(
         if "grad_sampler" in kwargs and self._use_gradient_buffering is not True:
             # set grad sampler, must be done before initialize in case pruning step
             # occurs on initialize epoch
-            grad_sampler = kwargs["grad_sampler"]
-            if not isinstance(grad_sampler, GradSampler):
-                raise ValueError(
-                    "grad_sampler must be an instance of the GradSampler class"
+            if (
+                "data_loader_builder" not in kwargs["grad_sampler"]
+                or "loss_function" not in kwargs["grad_sampler"]
+            ):
+                raise RuntimeError(
+                    "grad_sampler dict with data_loader_builder and loss_function "
+                    "must be provided to initialize GradSampler"
                 )
-            self._grad_sampler = grad_sampler
+            self._grad_sampler = GradSampler(
```

Comment:
Why not keep the same kwarg names as GradSampler(data_loader, loss_fn)? That way it's clearer to the user how to name their kwargs and you can pass GradSampler(**self._grad_sampler_kwargs) rather than passing them as args manually.

Reply:
Not sure if I understand the proposed idea, but `self._grad_sampler_kwargs` can't be used in the context `GradSampler(**self._grad_sampler_kwargs)` as it represents a dictionary of user-defined configs for gradient sampling procedure. `GradSampler` constructor expects an iterable dataloader and a callable loss function.